agent/php_user_instrument_wraprec_hash.h

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if LOOKUP_METHOD == LOOKUP_USE_WRAPREC_HASHMAP
+
+/* This hashmap will store pointers to wraprecs stored in the linked list.
+ * Since wraprecs are persistent (they're not destroyed between requests), 
+ * there's no need for hashmap value destructor. */
+
+typedef struct _nr_wraprec_hashmap_bucket_t {
+  struct _nr_wraprec_hashmap_bucket_t* prev;
+  struct _nr_wraprec_hashmap_bucket_t* next;
+  nruserfn_t *wraprec;
+} nr_wraprec_hashmap_bucket_t;
+
+struct _nr_wraprec_hashmap_t {
+  size_t hash_modulo;
+  nr_wraprec_hashmap_bucket_t** buckets;
+  size_t elements;
+};
+
+#ifdef DECLARE_nr_php_wraprec_hashmap_API
+extern nr_php_wraprec_hashmap_t* nr_php_wraprec_hashmap_create(void);
+#else
+nr_php_wraprec_hashmap_t* nr_php_wraprec_hashmap_create() {
+  nr_php_wraprec_hashmap_t* hashmap;
+
+  hashmap = (nr_php_wraprec_hashmap_t*)nr_malloc(sizeof(nr_php_wraprec_hashmap_t));
+    /*
+     * Encode the default value in one place: namely, here.
+     */
+  hashmap->hash_modulo = 257; //521 or 509 //53; // or 97 or other prime number
```

Comment:
The closer the number to the average the length of a PHP file, the better distribution of wraprecs between buckets and the shorter the list in the bucket. The number affects the memory used by buckets, i.e. `hash_modulo x sizeof( nr_wraprec_hashmap_bucket_t`).